src/main/java/com/joejoe2/chat/service/channel/GroupChannelServiceImpl.java

```diff
@@ -235,7 +235,7 @@ public GroupChannelProfile createChannel(String fromUserId, String name) throws
     User user =
         userRepository
```

Comment:
Extracting this into the method:
```java
private User findUserById(String fromUserId) throws UserDoesNotExist {
    return userRepository
            .findById(uuidValidator.validate(fromUserId))
            .orElseThrow(() -> UserDoesNotExist.ofId(fromUserId));
  }
```

it totally fine. By doing this we follow the DRY principle.

The same applied to the group.

Reply:
As a side note, we can pass UUID directly - no need to validate this one. It just makes things deliberately complex. But Engineering is about making things simple. So the simplest here is to eliminate this validation and pass UUID as it is.

Reply:
You're right, extract to a private method is better here. For validation part I will refactor in my free time (not in this branch 😔).